js/service/userMessageService.js

```diff
@@ -37,19 +37,34 @@ angular.module('afpClientApp').factory('userMessageService', [
     function createAccountArray(response) {
       var result = [];
 
-      angular.forEach(response, function (roles, accountName) {
-        var account = {
-          'AccountName': accountName,
-          'Roles': []
-        };
+      angular.forEach(response, function (details, accountName) {
+        var account = "";
+        if ("id" in details) {
+          account = {
+            'AccountName': accountName,
+            'Roles': [],
+            'AccountID': details['id']
+          };
 
-        angular.forEach(roles, function (role) {
-          account.Roles.push({
-            'Name': role,
-            'URLSuffix': accountName + '/' + role
+          angular.forEach(details['roles'], function (role) {
```

Comment:
Same here. Object key access is better written with the dot notation. `details.roles`